pkg/challenges/challenges.go

```diff
@@ -363,13 +408,16 @@ func isSpiffeIDAllowed(host, spiffeID string) bool {
 // isURISubjectAllowed compares the subject and issuer URIs,
 // returning an error if the scheme or the hostnames do not match
 func isURISubjectAllowed(subject, issuer *url.URL) error {
-	subjectHostname := subject.Hostname()
-	issuerHostname := issuer.Hostname()
-
 	if subject.Scheme != issuer.Scheme {
 		return fmt.Errorf("subject (%s) and issuer (%s) URI schemes do not match", subject.Scheme, issuer.Scheme)
 	}
 
+	return isDomainAllowed(subject.Hostname(), issuer.Hostname())
+}
+
+// isDomainAllowed compares two hostnames, returning an error if the
+// top-level and second-level domains do not match
+func isDomainAllowed(subjectHostname, issuerHostname string) error {
```

Comment:
Also the error messages talk about "subject URI" and "issuer URI" which to me doesn't seem like enough context—if you call `Username`, it implicitly fetches the config and it's an error in the *configuration* that it's complaining about, not really an error in what you passed it